cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -430,6 +431,23 @@ public synchronized void initAdjustment(LoadMonitor loadMonitor,
                                               requestedInterBrokerPartitionMovementConcurrency,
                                               requestedIntraBrokerPartitionMovementConcurrency,
                                               requestedLeadershipMovementConcurrency);
+      _started = true;
+    }
+
+    /**
+     * Concurrency should be reset after each execution is done.
+     */
+    public synchronized void clearAdjustment() {
+      _executionConcurrencyManager.reset();
+      _started = false;
```

Comment:
Could somebody be using `_executionConcurrencyManager` in a way that
1. check `_started` first
2. do something with the manager
3. it got reset in the middle
4. started mark as false

If that's so, either swap the 2 statements, or lock on the _started flag (by changing it to some other type suitable for such synchronization)

Reply:
Asking this because the code path that references `_started` here does not seem to be synced.

Reply:
Also the other way around, if we swap that: is it possible some code path would assume if it is false, then the `_executionConcurrencyManager` has been reset?

Thinking more about it, I'm leaning forward that the code path on this `_started` flag should be revisited.  Maybe an RW lock is needed.

Reply:
Good point! I think your are right that we should swap the 2 lines.

Reply:
> Also the other way around, if we swap that: is it possible some code path would assume if it is false, then the _executionConcurrencyManager has been reset?

Revisited the logic. Since ConcurrencyAdjuster and ConcurrencyManager are really different entities, one shouldn't make any assumptions that one is dependent on another. Thus RW lock is not needed here.